task/k8s/resources/resource_tx_pod.go

```diff
@@ -0,0 +1,146 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	k8s_apps "k8s.io/api/apps/v1"
+	k8s_core "k8s.io/api/core/v1"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	k8s_meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	"terraform-provider-iterative/task/common"
+	"terraform-provider-iterative/task/k8s/client"
+)
+
+// NewTransferPod creates a pod used to transfer data between the persistent storage and local.
+func NewTransferPod(client *client.Client, identifier common.Identifier, persistentVolumeClaim *PersistentVolumeClaim, permissionSet *PermissionSet, task common.Task) *TransferPod {
+	p := &TransferPod{
+		Client:     client,
+		Identifier: identifier.Long(),
+	}
+	p.Dependencies.PersistentVolumeClaim = persistentVolumeClaim
+	p.Dependencies.PermissionSet = permissionSet
+	p.Attributes.Task = task
+	return p
+}
+
+// TransferPod is a deployment with a single pod used to transfer data between the persistent volume claim and local.
+type TransferPod struct {
+	Client     *client.Client
+	Identifier string
+	Attributes struct {
+		Task      common.Task
+		Addresses []net.IP
+	}
+	Dependencies struct {
+		PersistentVolumeClaim *PersistentVolumeClaim
+		PermissionSet         *PermissionSet
+	}
+	Resource *k8s_apps.Deployment
+}
+
+// Create creates the transfer pod resource.
+func (p *TransferPod) Create(ctx context.Context) error {
+	if p.Attributes.Task.Environment.Directory == "" {
+		return fmt.Errorf("output directory not set")
+	}
+
+	volumeMounts := []k8s_core.VolumeMount{{
+		Name:      p.Identifier + "-pvc",
```

Comment:
Suffixes like `-pvc` are rather futile, given that Kubernetes resources are already classified into different collections. E.g. there won't be a name clash between a Pod named `example` and a namesake PersistentVolume.

Reply:
True. Thanks!

Reply:
Resolved with https://github.com/iterative/terraform-provider-iterative/pull/669/commits/97cb39e774d9d00790402afeaa544fd19f565a37